src/nvim/tui/tui.c

```diff
@@ -1813,6 +1813,14 @@ static const char *tui_tk_ti_getstr(const char *name, const char *value,
     if (value != NULL && strequal(stty_erase, value)) {
       return stty_erase[0] == DEL ? CTRL_H_STR : DEL_STR;
     }
+  } else if (strequal(name, "key_mouse")) {
+    DLOG("libtermkey:kmous=%s", value);
+    // libtermkey assumes mouse encoding protocol is always X10 in its
+    // terminfo driver (driver-ti.c), which is wrong. Don't use it if it's SGR
+    // encoding. XXX how about other encodings (UTF-8, URXVT...)?
+    if (value != NULL && strequal("\033[<", value)) {
+      return NULL;
```

Comment:
Shouldn't we return NULL always? Why do we care what `value` is?

Reply:
Oops, forgot to mention another tricky thing in libtermkey: the CSI driver can handle SGR and URXVR protocols but not the X10 protocol.

https://github.com/neovim/libtermkey/blob/master/driver-csi.c#L162

neovim already forces the SGR protocol (1006), so it's safe to skip the value check. I put it here so that neovim won't break in the future if it wants a different mouse encoding protocol.

(According to the author of ncurses, some terminals do not implement 1006 protocol correctly [1])

[1] https://raw.githubusercontent.com/ThomasDickey/ncurses-snapshots/master/misc/terminfo.src